File: Programming/Python/Roman_to_Integer/roman_to_integer.py
def get_integer_from_roman(roman):
	dictionary = {
		'I': 1,
		'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
	}
	prev, curr, ans = 0, 0, 0
	for char in roman:
		curr = dictionary.get(char)
		if not curr:
			return "Does not exist!"
		ans += curr - 2*prev if curr > prev else curr
		prev = curr
	return ans


def main():
	roman = input('Enter the roman number!\n')
	integer = get_integer_from_roman(roman)
	if integer == "Does not exist!":
		print(integer)
		print('Please give a correct number or in the range of 1 to 3999')
		return
	print(roman, '-->', integer)

if __name__ == '__main__':
	main()

File: Programming/Golang/Roman_to_Integer/roman_to_integer.go
package main
import (
	"fmt"
)

func get_integer_from_roman(roman string) int {
	ma := map[string]int{
		"I": 1,
		"V": 5,
        "X": 10,
        "L": 50,
        "C": 100,
        "D": 500,
        "M": 1000,
	}
	var prev, curr, ans int
	for _, char := range roman {
		curr = ma[string(char)]
		if curr > prev {
			ans += curr - 2*prev
		} else {
			ans += curr
		}
		prev = curr
	}
	return ans
}

func main() {
	var roman string
	fmt.Println("Enter the roman number!\n")
	fmt.Scan(&roman)
	var integer int
	integer = get_integer_from_roman(roman)
	fmt.Println(roman, "-->", integer)
}